### DISCUSSION TITLE
CXT batteries

### DISCUSSION THREAD

**Discussion by eroyee** [mnh-jansson/open-battery-information#88] [Makita]
It would appear there **is** a Makita tester for the CXT range of batteries, a BTC06. I found it somewhat difficult to get information about, but this is the translated text from a Dutch site:

"Using the battery tester, the condition of a CXT battery can be read within 5 seconds. This way you can see whether the battery is in perfect condition or why it is not. It is possible to read the following things: voltage of the entire battery, voltage of a cell, remaining battery capacity, error message readout function and temperature of the battery/cells. Power supply via 4 AA (LR6) batteries or USB adapter DEBAD05 in combination with Makita battery. The following CXT Li-ion batteries can be tested: BL1015, BL1016, BL1020B, BL1021B, BL1040B, BL1041B"

No mention of number of cycles, so it's unclear whether the battery records that detail or not.

The CXT battery appears to have 5 pins, at this stage I've not found any data on what they each do, nor decent photographs of the board. What I have been able to establish is that there are at least two different board types, and at least two different cell types (18650 and 21700). The 21700 cell type appears to have quite a different board from the others.

**Comment by mnh-jansson** [comment]
I have a CXT battery now and it only has the cell connections and temperature sensor to the 5 pins, the battery indicator is seperate and there is only VBAT going to this PCB. So i assume the tester only analyses the cells and temperature when on the charger and then determines the error code.

Nothing that makes sense to provide here i think

**Comment by eroyee** [comment]
We may be talking at cross-purposes here but from what I could determine the BTC06 is a completely separate unit that doesn't utilise the charger, so it must get the data from the battery somehow? 

FWIW I knocked up this graphic, given you've got a battery you may be able to confirm its veracity:

![Image](https://github.com/user-attachments/assets/fb8a1b35-7e41-4a2b-95ad-5b7711a22490)

I recall wondering if the communication protocol for this battery could be similar to that of the 40V battery? No doubt you're aware but for the completeness of this post, and anyone landing here, there are two 40V Github projects [here](https://github.com/Malvineous/makita-xgt-serial) and [here](https://github.com/twaymouth/XGT-Tester), which may be a useful reference for anyone wanting to work on communicating with the CXT battery.

**Comment by mnh-jansson** [comment]
The tester would most likely get the information in the same way as the charger. I understand that the BTC06 is a seperate unit, it would have to measure the cell voltages and temperature through the connector in the same way the charger would.

I also made a simple schematic of the PCB. 
![Image](https://github.com/user-attachments/assets/518ccb76-266a-46dd-a4f6-835ca9c31ac1)

**Comment by eroyee** [comment]
That's interesting (and very useful). It would seem from this there's no data held within the battery that's externally available, yet the BTC06 detail suggests that _"remaining battery capacity, error message readout"_ are obtainable. 

I guess it must be roughly determining these parameters from the existing pinouts via rapid cell assessment, but for capacity at least it lacks the accuracy of a decent coulomb counter. 

Anyway it would seem my surmise gave rather more intelligence to this battery than was warranted, which I suppose makes it easy to construct a tester - although I doubt it's worth it, a voltmeter is going to give most of what one wants, other than capacity. 

One other thing though, do you know if these batteries will lock up as the others do?

**Comment by mnh-jansson** [comment]
"Remaining battery capacity" i would call bullshit on, i suspect they only do a voltage curve estimate. Error message readout is, i guess somewhat valid claim, but that's also just looking at the cells and temperature. It would be dumb if they had any BMS to not track number of charges, and since they dont show that in BTC06, i'm pretty sure it does not exist.

I agree with you, a tester is possible, but multimeter would give the exact same information. 

These batteries have no way of locking up other than actual hardware faults. The way god intended :)

There is a video on Youtube showcasing the tester:
https://www.youtube.com/watch?v=f3ipTTWXOs4

**Comment by mktfn** [comment]
I have a BTC06 and am willing to contribute in whatever way I can.  If these batteries don't software lock, it would still be nice to have the information that's provided by the tester, such as cell health/balance/voltage, etc.  There is a very small capacity CXT battery which doesn't have a battery level indicator, so it's less convenient to get the voltage of these batteries.  I got the BTC06 specifically to monitor the health of these batteries, as well as my other CXT batteries.  It is indeed troublesome to get hands on.

Let me know if you have any questions about the tester as well.  I can provide what information I can.

I also have the BTC04 for the LXT batteries, with the XGT adapter, and XGT batteries, if there is something I can contribute there.
